spec/lib/jobs/dell-wsman-inventory-spec.js

```diff
@@ -0,0 +1,269 @@
+            configuration.get.withArgs('dell').returns(dellConfigs);
+            expect(wsmanInventoryJob._initJob()).to.be.rejectedWith('No target IP address.');
+        });
+
+        it('should handle the case of incorrect node type', function() {
+            wsmanInventoryJob.nodeType = 'unknown';
+            wsmanBaseJob.prototype.checkOBM.resolves(obm);
+            wsmanBaseJob.prototype.getIpAddress.resolves('172.31.128.73');
+            configuration.get.withArgs('dell').returns(dellConfigs);
+            expect(wsmanInventoryJob._initJob()).to.be.rejectedWith('Inventory collection for node type (unknown) is not implemented.');
+        });
+
+        it('should init job correctly', function() {
+            wsmanInventoryJob.nodeType = 'compute';
+            wsmanBaseJob.prototype.checkOBM.resolves(obm);
+            wsmanBaseJob.prototype.getIpAddress.resolves('172.31.128.73');
+            configuration.get.withArgs('dell').returns(dellConfigs);
+            return wsmanInventoryJob._initJob().then(function() {
+                expect(wsmanInventoryJob.retrys.length === 4);
+            });
+        });
+    });
+
+    describe('inventoryCallback', function() {
+        it('should handle the callback from smi service', function() {
+            var data = {
+                type: 'nics',
+                data: {
+                    body: 'fake nic json string'
+                }
+            };
+            wsmanInventoryJob.nodeId = '59c4720870bf1d7c172930db';
+            WsmanInventoryJobClass.prototype.handleAsyncResponse.resolves();
+            WsmanInventoryJobClass.prototype._handleAsyncRequest.resolves();
+            return wsmanInventoryJob.inventoryCallback(data).then(function() {
+                expect(WsmanInventoryJobClass.prototype._handleAsyncRequest).to.have.been.calledOnce;
+            });
+        });
+    });
+
+    describe('handleAsyncResponse', function() {
+        var data = {
+            id: 0,
+            fqdd: 'NIC.Embedded.2-1-1',
+            productName: 'Intel(R) Ethernet 10G X520 LOM - 7C:D3:0A:B0:52:A0'
+        };
+        var catalog = {
+            id: '59c4720870bf1d7c172930db',
+            source: 'software',
+            data: {
+                componentType: {
+                    value: 'FRMW',
+                    otherAttributes: {}
+                },
+                elementName: {
+                    value: 'Integrated Dell Remote Access Controller',
+                    otherAttributes: {}
+                }
+            }
+        };
+
+        beforeEach(function() {
+            loggerSpy.reset();
+            WsmanInventoryJobClass.prototype.handleAsyncResponse.restore();
+        });
+
+        it('should handle the case that no catalog created.', function() {
+            return wsmanInventoryJob.handleAsyncResponse({}, 'software').then(function() {
+                expect(loggerSpy).to.have.been.calledOnce;
```

Comment:
It can't identify which case is running.

Reply:
Done. Add new assertion to enhance validation.